projects/vdk-plugins/vdk-jupyter/vdk-jupyterlab-extension/src/components/ConvertJobToNotebook.tsx

```diff
@@ -81,13 +81,14 @@ export async function showConvertJobToNotebookDialog(
     });
     if (confirmation.button.accept) {
       statusButton?.show('Convert', jobData.get(VdkOption.PATH)!);
-      let { message, status } = await jobConvertToNotebookRequest();
-      if (status) {
-        const transformjobResult = JSON.parse(message);
-        notebookContent = initializeNotebookContent(
-          transformjobResult['code_structure'],
-          transformjobResult['removed_files']
-        );
+      let { message, isSuccessful } = await jobConvertToNotebookRequest();
+      if (isSuccessful) {
+        if (typeof message !== 'string') {
```

Comment:
Nit: It might be worth merging this condition with the above one, e.g.

```
if (isSuccessful && typeof message !== 'string') {
// do stuff
}
```

Reply:
Also, it's not really clear why we check the message type.

Reply:
I think when we introduce the 2 types Paul proposed, it will be more clear.

Reply:
Actually, I will add a comment to explain why we do this.